File: Current File Name/current_file_name.py
#
#   Ensures the current file name is as expected, Python
#   Module written by Brandon Arvanaghi 
#   Website: arvanaghi.com 
#   Twitter: @arvanaghi
#

import os
import sys

expectedFileName = ""

if len(sys.argv) != 2:
  print("You must provide a file name to check for.")
  exit(1)
else:
  expectedFileName = sys.argv[1]

if os.path.basename(__file__) == expectedFileName:
  print("The file name is {} as expected. Proceed!".format(expectedFileName))
else:
  print("The file name {0}, not {1} as expected. Do not proceed.".format(os.path.basename(__file__), expectedFileName))

File: Current File Name/current_file_name.go
/*
    Ensures the current file name is as expected, Go
    Module written by Brandon Arvanaghi 
    Website: arvanaghi.com 
    Twitter: @arvanaghi
*/

package main

import (
  "fmt"
  "os"
  "path/filepath"
)

func main() {

	expectedName := ""

	if (len(os.Args) != 2) {
		fmt.Println("You must provide a file name to check for.")
		os.Exit(1)
	} else {
		expectedName = os.Args[1]
	}

	actualName := filepath.Base(os.Args[0])

	if actualName == expectedName {
		fmt.Printf("The file name is %s as expected. Proceed!\n", actualName)
	} else {
		fmt.Printf("The file name %s, not %s as expected. Do not proceed.\n", actualName, expectedName)
	}

}